lib/collaborative_filtering.py

```diff
@@ -113,6 +116,35 @@ def train(self, item_vecs=None):
             self.fold_train_indices, self.fold_test_indices = self.evaluator.get_kfold_indices()
             return self.train_k_fold(item_vecs)
 
+    def build_confidence_matrix(self, index, type='user'):
+        """
+        Builds a confidence matrix
+        """
+
+        if(type == 'user'):
+            shape = self.item_vecs.shape[0]
+        else:
+            shape = self.user_vecs.shape[0]
+        
+        confidence = numpy.array([0.1] * shape)
+        for i in range(len(confidence)):
+            if(type == 'user'):
+                if(self.ratings[index][i] == 1):
+                    confidence[i] = 1
+            else:
+                if(self.ratings[i][index] == 1):
+                    confidence[i] = 1
+
+        return confidence
+
+    def scale_matrix(self, matrix, vector):
```

Comment:
If the matrix and vector are both ndarray, this function can just be replaced by (matrix * vector)